impl/py/pika_id/pika.py

```diff
@@ -0,0 +1,205 @@
+    - `node_id`: The node ID to use for the Snowflake generator. Defaults to 0.
+    - `suppress_prefix_warnings`: Whether to suppress warnings when an unregistered prefix is used. Defaults to False.
+
+    ```py
+    from pika import Pika, PikaPrefixDefinition, PikaInitializationOptions
+
+    prefixes = [
+        PikaPrefixDefinition(prefix="u", description="User ID"),
+        PikaPrefixDefinition(prefix="g", description="Group ID"),
+        PikaPrefixDefinition(prefix="c", description="Channel ID"),
+    ]
+
+    pika = Pika(prefixes, epoch=1640995200000, node_id=0, suppress_prefix_warnings=False)
+    ```
+    """
+    prefixes: Dict[str, PikaPrefixDefinition[P]]
+    snowflake: Snowflake
+    suppress_prefix_warnings: bool
+
+    def __init__(
+        self, 
+        prefixes: List['PikaPrefixDefinition[P]'], 
+        epoch: EpochResolvable = 1640995200000, 
+        node_id: int = 0, 
+        suppress_prefix_warnings: bool = False, 
+    ):
+        self.snowflake = Snowflake(epoch, node_id)
+        self.suppress_prefix_warnings = suppress_prefix_warnings
+        self.prefixes = { prefix.prefix : prefix for prefix in prefixes }
+
+    def validate(self, maybe_id: str, expect_prefix: P = None) -> bool:
+        parts = maybe_id.split('_')
+        tail = parts[-1]
+        prefix = '_'.join(parts[:-1])
+        
+        if not tail:
+            return False
+        
+        if expect_prefix and prefix != expect_prefix:
+            return False
+        
+        if expect_prefix:
+            return prefix == expect_prefix
+        
+        print(prefix, self.prefixes)
+
+        return prefix in self.prefixes
+
+    def gen(self, prefix: P) -> str:
+        """
+        Generates a Pika ID with the given prefix. If the prefix is not registered, a warning will be printed.
+
+        Args:
+            prefix (P): The prefix to use for the ID.
+
+        Returns:
+            str: The generated ID.
+
+        Examples:
+            ```py
+            from pika import Pika, PikaPrefixDefinition
+
+            prefixes = [
+                PikaPrefixDefinition(prefix="u", description="User ID"),
+                PikaPrefixDefinition(prefix="g", description="Group ID"),
+                PikaPrefixDefinition(prefix="c", description="Channel ID"),
+            ]
+
+            pika = Pika(prefixes)
+
+            user_id = pika.gen("u")
+            ```
+        """
+
+        if prefix not in self.prefixes and not self.suppress_prefix_warnings:
+            print(f"Warning: Unregistered prefix ({prefix}) was used.")
+        
+        snowflake = self.snowflake.gen()  # Assuming Snowflake has a gen() method
+        
+        prefix_definition = self.prefixes.get(prefix)
+        secure_prefix = f"s_{uuid.uuid4().hex}_" if prefix_definition and prefix_definition.secure else ''
+        
+        tail = base64.urlsafe_b64encode(f"{secure_prefix}{snowflake}".encode()).decode()
```

Comment:
Could you strip the padding from the b64encode? (ig just `rstrip('=')`) Pikas are meant to be alphanumeric

Reply:
Ofc, will take a look sometime this week :)